src/Arriba/Arriba/ArribaProgram.cs

```diff
@@ -25,20 +25,24 @@ public static async Task<int> Run<T>(Func<Task<int>> program)
 
         private static async Task ArribaRun<T>(Func<Task> program)
         {
+            TimeSpan duration;
             using (ArribaLogs.EnableConsoleOutput())
             {
                 try
                 {
                     ArribaEventSource.Service.Starting<T>();
-                    await program();
+                    duration = await ArribaEventSource.Time(async () =>
+                    {
+                        await program();
+                    });
```

Comment:
This breaks async as you are using a void delegate which means this code path will not what what it is intended